### PR TITLE
Updates to ordering definitions

### PR DESCRIPTION
**Pull request by shefty**
Eliminates ordering fields from fi_ep_attr.  Apps will use rx/tx attr instead.

Clarify ordering definitions for TX/RX.

Add data ordering flag

### CODE REVIEW

**Comment by shefty** [comment]
@sayantansur @patrickmacarthur @goodell 
Please review that these changes make sense.

File: man/fi_endpoint.3.md [hunk 1]
```diff
@@ -714,50 +792,186 @@ order values are:
...
+*FI_ORDER_SAS*
+: Send after send.  If set, message send operations, including tagged
+  sends, are processed in the order that they were transmitted relative to other
+  message send.  If not set, message sends may be processed out of
+  order from their submission.
+
+## comp_order - Completion Ordering
+
+For a description of message ordering, see the msg_order field in
+the _Transmit Context Attribute_ section.
```
Comment (patrickmacarthur) [inline man/fi_endpoint.3.md]: Minor nit: I think this should say completion ordering and `comp_order` instead.

**Comment by patrickmacarthur** [comment]
Other than my one comment, looks good to me.

File: man/fi_endpoint.3.md (hunk 1 above)
Comment (shefty) [inline man/fi_endpoint.3.md]: I'm assuming that you're referring to the order here.  The format matches the syntax used in other places in the file, where the field name comes first, followed by the description.  I can go either way with that, but all of the man pages would need to be updated for consistency.

Btw, thanks for pointing out that I left this series unmerged.

Comment (patrickmacarthur) [inline man/fi_endpoint.3.md]: I was referring to the paragraph underneath the completion ordering heading.  GitHub just gave lots of irrelevant context.

On March 31, 2015 6:48:06 PM EDT, Sean Hefty <notifications@github.com> wrote:
>> +*FI_ORDER_SAW*
>> +: Send after write.  If set, message send operations, including
>tagged
>> +  sends, are processed in the order that they were transmitted
>relative to RMA
>> +  and atomic write operations.  If not set, message sends may be
>processed ahead
>> +  of RMA and atomic writes.
>> +
>> +*FI_ORDER_SAS*
>> +: Send after send.  If set, message send operations, including
>tagged
>> +  sends, are processed in the order that they were transmitted
>relative to other
>> +  message send.  If not set, message sends may be processed out of
>> +  order from their submission.
>> +
>> +## comp_order - Completion Ordering
>> +
>> +For a description of message ordering, see the msg_order field in
>> +the _Transmit Context Attribute_ section.
>
>I'm assuming that you're referring to the order here.  The format
>matches the syntax used in other places in the file, where the field
>name comes first, followed by the description.  I can go either way
>with that, but all of the man pages would need to be updated for
>consistency.
>
>Btw, thanks for pointing out that I left this series unmerged.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/ofiwg/libfabric/pull/850/files#r27530884

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

Comment (goodell) [inline man/fi_endpoint.3.md]: I think he's saying that:
```diff
+For a description of message ordering, see the msg_order field in
+the _Transmit Context Attribute_ section.
```

Should be:
```diff
+For a description of completion ordering, see the comp_order field in
+the _Transmit Context Attribute_ section.
```

**Comment by goodell** [comment]
Other than Patrick's comment, :+1:

File: man/fi_endpoint.3.md (hunk 1 above)
Comment (shefty) [inline man/fi_endpoint.3.md]: Thanks - fixed and will push updated patch shortly.  This isn't quite a nit IMO.

**Pull request merged**